From 70f44f3fa819e2c13726895a2c1e3c5e8fa51eef Mon Sep 17 00:00:00 2001
From: Andrew Jeffery <andrew@aj.id.au>
Date: Mon, 21 Mar 2016 13:49:47 +1030
Subject: [PATCH] arch/arm: Add qemupalmetto dts with Cadence GEM support

Describes a machine that is as similar to the Palmetto BMC, primarily for
running under QEMU. Relaxing the constraint of booting the exact Palmetto
BMC hardware in QEMU allows us to integrate existing device models to expand
the machine's capability without huge effort.

In this case, the Cadence GEM controller is integrated to provide networking.

Signed-off-by: Andrew Jeffery <andrew@aj.id.au>
---
 arch/arm/boot/dts/aspeed-bmc-opp-qemupalmetto.dts | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)
 create mode 100644 arch/arm/boot/dts/aspeed-bmc-opp-qemupalmetto.dts

diff --git a/arch/arm/boot/dts/aspeed-bmc-opp-qemupalmetto.dts b/arch/arm/boot/dts/aspeed-bmc-opp-qemupalmetto.dts
new file mode 100644
index 0000000..b0915cd
--- /dev/null
+++ b/arch/arm/boot/dts/aspeed-bmc-opp-qemupalmetto.dts
@@ -0,0 +1,16 @@
+#include "aspeed-bmc-opp-palmetto.dts"
+
+/ {
+	ahb {
+		mac1: ethernet@1e680000 {
+			compatible = "cdns,gem";
+			reg = <0x1e680000 0x1000>;
+			interrupts = <3>;
+			clock-names = "pclk", "hclk", "tx_clk";
+			clocks = <&clk_apb>, <&clk_apb>, <&clk_apb>;
+			status = "okay";
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+	};
+};
-- 
2.5.0

